javaslang/src/main/java/javaslang/collection/Arrays2.java

```diff
@@ -13,10 +13,49 @@
  * @author Pap Lőrinc
  * @since 3.0.0
  */
+@SuppressWarnings({"unchecked", "SuspiciousArrayCast"})
 final class Arrays2 { // TODO reuse these in `Array` also
```

Comment:
Valid question, because we use `java.util.Arrays` from the same classes and I didn't want to fully qualify it.
I think the internal `Collections` should also be renamed to `Collections2`  ([Guava](https://google.github.io/guava/releases/snapshot/api/docs/com/google/common/collect/Collections2.html) style), or `SlangCollections`, since we definitely need both from the same namespace.